src/test/java/net/snowflake/client/jdbc/ResultSet0IT.java

```diff
@@ -23,79 +23,78 @@ public class ResultSet0IT extends BaseJDBCTest {
 
   public Connection init(int injectSocketTimeout) throws SQLException {
     Connection connection = BaseJDBCTest.getConnection(injectSocketTimeout);
-
-    Statement statement = connection.createStatement();
-    statement.execute(
-        "alter session set "
-            + "TIMEZONE='America/Los_Angeles',"
-            + "TIMESTAMP_TYPE_MAPPING='TIMESTAMP_LTZ',"
-            + "TIMESTAMP_OUTPUT_FORMAT='DY, DD MON YYYY HH24:MI:SS TZHTZM',"
-            + "TIMESTAMP_TZ_OUTPUT_FORMAT='DY, DD MON YYYY HH24:MI:SS TZHTZM',"
-            + "TIMESTAMP_LTZ_OUTPUT_FORMAT='DY, DD MON YYYY HH24:MI:SS TZHTZM',"
-            + "TIMESTAMP_NTZ_OUTPUT_FORMAT='DY, DD MON YYYY HH24:MI:SS TZHTZM'");
-    statement.close();
+    try (Statement statement = connection.createStatement()) {
+      statement.execute(
+          "alter session set "
+              + "TIMEZONE='America/Los_Angeles',"
+              + "TIMESTAMP_TYPE_MAPPING='TIMESTAMP_LTZ',"
+              + "TIMESTAMP_OUTPUT_FORMAT='DY, DD MON YYYY HH24:MI:SS TZHTZM',"
+              + "TIMESTAMP_TZ_OUTPUT_FORMAT='DY, DD MON YYYY HH24:MI:SS TZHTZM',"
+              + "TIMESTAMP_LTZ_OUTPUT_FORMAT='DY, DD MON YYYY HH24:MI:SS TZHTZM',"
+              + "TIMESTAMP_NTZ_OUTPUT_FORMAT='DY, DD MON YYYY HH24:MI:SS TZHTZM'");
+    }
     return connection;
   }
 
   public Connection init() throws SQLException {
     Connection conn = BaseJDBCTest.getConnection(BaseJDBCTest.DONT_INJECT_SOCKET_TIMEOUT);
```

Comment:
add to try

Reply:
This method returns the connection instance. Should not be closed.